File: commons-client/src/test/java/test/pl/wrzasq/commons/client/interceptor/BearerTokenAuthorizerTest.java
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2017 - 2019 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package test.pl.wrzasq.commons.client.interceptor;

import java.util.ArrayList;

import feign.RequestTemplate;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpHeaders;
import pl.wrzasq.commons.client.interceptor.BearerTokenAuthorizer;

public class BearerTokenAuthorizerTest {
    @Test
    public void apply() {
        var token = "test";
        var template = new RequestTemplate();

        new BearerTokenAuthorizer(token).apply(template);

        var values = template.headers().get(HttpHeaders.AUTHORIZATION);
        Assertions.assertEquals(
            "Bearer test",
            new ArrayList<>(values).get(0),
            "BearerTokenAuthorizer.apply() should set Authorization HTTP header."
        );
    }
}

File: commons-client/src/test/kotlin/test/pl/wrzasq/commons/client/interceptor/BearerTokenAuthorizerTest.kt
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2017 - 2021 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package test.pl.wrzasq.commons.client.interceptor

import feign.RequestTemplate
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import pl.wrzasq.commons.client.interceptor.BearerTokenAuthorizer
import pl.wrzasq.commons.client.interceptor.HEADER_NAME_AUTHORIZATION

class BearerTokenAuthorizerTest {
    @Test
    fun apply() {
        val token = "test"
        val template = RequestTemplate()
        BearerTokenAuthorizer(token).apply(template)
        val values = template.headers()[HEADER_NAME_AUTHORIZATION]
        assertEquals("Bearer test", values?.first())
    }
}
